### PR TITLE
Acl cleanup

### PR DESCRIPTION
**Pull request by megoth**
Some work that I started doing for cleaning up the ACL module. It is not complete, but I think it adds value to the code base.

I hope the added matchers are reasonable additions to the test suite.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
 module.exports = {
-  verbose: true
+  verbose: true,
+  setupFilesAfterEnv: [
+    './test/setup.ts'
+  ]
```
Comment (michielbdejong) [inline jest.config.js]: Seems your change from https://github.com/solid/solid-ui/pull/280 accidentally made it in here? Please remove.

File: test/setup.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { toContainGraph } from './custom-matchers/toContainGraph'
+import { toEqualGraph } from './custom-matchers/toEqualGraph'
+
+// adding custom matchers
+expect.extend({
+  toContainGraph,
+  toEqualGraph
+})
```
Comment (michielbdejong) [inline test/setup.ts]: OK to add this file, but please call it from individual test files, not from jest config.

File: test/unit/acl/acl.test.ts [hunk 3]
```diff
@@ -22,230 +22,374 @@ import {
...
+// })
+//
+// describe('ACLToString', () => {
+//   it('exists', () => {
+//     expect(ACLToString).toBeInstanceOf(Function)
+//   })
+//   it('runs', () => {
+//     expect(ACLToString({} as AgentMapMap)).toEqual('{}')
+//   })
+// })
```
Comment (michielbdejong) [inline test/unit/acl/acl.test.ts]: Why did you comment this out? Committed by accident?

File: test/unit/acl/acl.test.ts [hunk 4]
```diff
@@ -22,230 +22,374 @@ import {
...
+// describe('fixIndividualCardACL', () => {
+//   it('exists', () => {
+//     expect(fixIndividualCardACL).toBeInstanceOf(Function)
+//   })
+//   it('runs', () => {
+//     expect(fixIndividualCardACL(sym('https://test.test#'), () => {
+//     }, () => {
+//     })).toEqual(undefined)
+//   })
+// })
```
Comment (michielbdejong) [inline test/unit/acl/acl.test.ts]: Please uncomment and where necessary use .skip

File: test/unit/acl/acl.test.ts [hunk 5]
```diff
@@ -22,230 +22,374 @@ import {
...
+//       createLiveStore(),
+//       sym('https://test.test#'),
+//       {} as AgentMapMap,
+//       sym('https://test.test#'),
+//       () => {
+//       }
+//     )).toEqual(undefined)
+//   })
+// })
+
```
Comment (michielbdejong) [inline test/unit/acl/acl.test.ts]: Please uncomment and where necessary use .skip

**Comment by michielbdejong** [comment]
Not sure why Trellis is yellow, it shows green when you follow the link.

File: test/setup.ts (hunk 2 above)
Comment (megoth) [reply test/setup.ts]: I'm uncertain of how much value it is to make this opt-in? Yes, there's the problem of discoverability, perhaps I should add a section to developer docs describing these custom matchers instead?

File: jest.config.js (hunk 1 above)
Comment (megoth) [reply jest.config.js]: See https://github.com/solid/solid-ui/pull/283#discussion_r397188875

File: test/unit/acl/acl.test.ts (hunk 3 above)
Comment (megoth) [reply test/unit/acl/acl.test.ts]: Sorry about that, forgot to readd tests that I didn't have time to complete. I've readded the tests in my latest commit.

File: test/setup.ts (hunk 2 above)
Comment (michielbdejong) [reply test/setup.ts]: See https://github.com/solid/solid-ui/pull/285

Comment (megoth) [reply test/setup.ts]: Hence my point on discoverability. I don't think we should make custom matchers opt-in though, they should always be available to all tests.

Comment (michielbdejong) [reply test/setup.ts]: We're commenting on a commit that doesn't exist anymore? My original comment was about silencing the debug messages, I didn't intend to say anything about custom matchers.

Sounds like yeah, these matchers should always be available to all tests.

Comment (megoth) [reply test/setup.ts]: Then you commented the wrong place ;) Or maybe we're seeing different things:

Screen Shot 2020-03-25 at 17 54 39

Ok, but then this PR should be ok, and I'll merge it as is.

**Pull request merged**